packages/expo-linking/src/Linking.ts

```diff
@@ -24,6 +23,10 @@ function usesCustomScheme(): boolean {
 
 function getHostUri(): string {
   if (!manifest.hostUri && !usesCustomScheme()) {
+    if (!Constants.linkingUri) {
+      // this is a bare workflow app
+      return '';
```

Comment:
i would change the `getHostUri` function to look like this, for clarity:

```js
function getHostUri(): string | null {
  if (manifest.hostUri) {
    return manifest.hostUri;
  } else if (!manifest.hostUri && !usesCustomScheme()) {
    // we're probably not using up-to-date xdl, so just fake it for now
    // we have to remove the /--/ on the end since this will be inserted again later
    return removeScheme(Constants.linkingUri).replace(/\/--($|\/.*$)/, '');
  } else {
    return null;
  }
}
```

it looks like the rest of the code already expects `getHostUri()` to return `string | null` anyhow, eg: in `parse`:

```
const hostUri = getHostUri() || '';
```